package/gluon-web/luasrc/usr/lib/lua/gluon/web/dispatcher.lua

```diff
@@ -184,9 +184,15 @@ local function dispatch(config, http, request)
 		return
 	end
 
-	http:parse_input(node.filehandler)
+	local ok, err = pcall(http.parse_input, http, node.filehandler)
+	if not ok then
+		http:status(400, "Bad request")
+		http:prepare_content("text/plain")
+		http:write(err)
```

Comment:
``` suggestion
		http:write(err .. '\n')
```

A minor thing. We could add a newline here. This looks nicer, e.g. if you are using curl. Otherwise my terminal looks like this:

```
/usr/lib/lua/gluon/web/http/protocol.lua:189: POST request with invalid Origin header[lemoer@orange ~]$ 
```

Reply:
Good idea. I used `\r\n` (which may or may not be better than `\n` - FWIW, it's what's used in the HTTP header) and also added it to the other `text/plain` error output.